x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/test_pipeline.tsx

```diff
@@ -151,9 +151,31 @@ export const TestPipeline: React.FC = () => {
                     </strong>
                   </p>
                 </EuiText>
-                <EuiCodeBlock fontSize="m" isCopyable language="json" paddingSize="m">
-                  {`[{"_index":"index","_id":"id","_source":{"${sourceField}":"bar"}}]`}
-                </EuiCodeBlock>
+                {fieldMappings && (
+                  <EuiCodeBlock fontSize="m" isCopyable language="json" paddingSize="m">
+                    {`[{"_index":"index","_id":"id","_source":{${fieldMappings
+                      .map(
```

Comment:
Code duplication for the component - could we move the decision logic inside the evaluation of `_source`? (Not sure if this is the correct syntax.)
```
<EuiCodeBlock fontSize="m" isCopyable language="json" paddingSize="m">
                    {`[{"_index":"index","_id":"id","_source":{ 
                       $fieldMappings &&  $fieldMappings.map(...)
                       $sourceField && "${sourceField}": ...
                    }
</EuiCodeBlock>
```